tv/tv.cpp

```diff
@@ -537,12 +539,16 @@ struct TVInitPass : public llvm::PassInfoMixin<TVInitPass> {
 
 struct TVFinalizePass : public llvm::PassInfoMixin<TVFinalizePass> {
   static bool finalized;
-  llvm::PreservedAnalyses run(llvm::Function &F,
-                              llvm::FunctionAnalysisManager &FAM) {
+  static void finalize(llvm::Module &M) {
     if (!finalized) {
       finalized = true;
```

Comment:
is this finalized flag still needed?

Reply:
Yes, because finalize can be potentially called twice:
registerOptimizerLastEPCallback's `MPM.addPass(TVFinalizePass());`, and TVNewPass's `TVFinalizePass::finalize(M);`.
I saw a few LLVM unit tests failing from this.

Reply:
It happens when a default pipeline is used, but it is now omitted after the updates in the python script, so this flag seems unnecessary.
However, a user may unconsciously use sth like `-passes=module(tv),default<O2>,module(tv)`, which will invoke the double finalization again.
Should we remove this finalization flag anyway?

Reply:
let's keep it then. better safe than crash!